utils/run_pytest.py

```diff
@@ -3,72 +3,131 @@
 
 BASE_REPO = "repos"
-REPOS = ["common", "containerization", "upgrade"]
+TMP_BASE_REPO = "tmp_" + BASE_REPO
 
-def get_actor_tests(actor_name):
-    """ Checks if the actors exists and has tests. Reports errors otherwise.
 
-    Returns (list of actor test directories | None, status message)
-    """
-    for repo in REPOS:
-        repository = scan_repo(BASE_REPO + "/" + repo)
-        repository.load()
-        actor = repository.lookup_actor(actor_name)
-        if not actor:
-            return (None, "Error: cannot find actor \"{ACTOR_NAME}\"!".format(ACTOR_NAME=actor_name))
-        if not actor.tests:
-            return (None, "Error: actor \"{ACTOR_NAME}\" is missing tests!".format(ACTOR_NAME=actor_name))
-        testdirs = []
-        for testdir in actor.tests:
-            testdirs.append(os.path.join(BASE_REPO, repo, actor.directory, testdir))
-        return (testdirs, "{ACTOR_NAME} tests found!".format(ACTOR_NAME=actor_name))
-    return (None, "Error: WAT?!")
-
-def print_pretty(msg):
-    """ Prints msg in a pretty way.
+def snactor_register(path):
+    """ Snactor registers all repos in @path.
     """
-    print("+{STUFF}+\n| {MSG} |\n+{STUFF}+".format(STUFF="-"*(len(msg)+2), MSG=msg))
+    cmd = "snactor repo find --path {PATH}".format(PATH=path)
+    try:
+        return subprocess.check_output(cmd, shell=True)
+    except OSError as exc:
+        sys.stderr.write(str(exc) + '\n')
+        return None
 
 if __name__ == "__main__":
-    pytest_cmd = ["pytest", "-v", BASE_REPO]
+    pytest_cmd = ["pytest", "-v"]
+    pytest_cmd.append(TMP_BASE_REPO)
 
     parser = argparse.ArgumentParser()
     parser.add_argument("--actor", help="name of the actor for which to run tests")
     parser.add_argument("--report", help="filepath where to save report")
     args = parser.parse_args()
 
+    # User wants to test single actor specified in the args.actor.
     if args.actor:
-        testdirs, status = get_actor_tests(args.actor)
-        print_pretty(status)
-        if not testdirs:
-            sys.exit(1)
-        formatted_testdirs = " ".join(testdirs)
-        pytest_cmd += ["{TESTDIRS}".format(TESTDIRS=formatted_testdirs)]
+        # NOTE: ACTOR flag is disabled. Check module docstring for more info.
+        logger.critical(" ACTOR flag is currently disabled!")
+        sys.exit(1)
 
+    # User wants to output xml report for junit.
     if args.report:
         pytest_cmd += ["--junit-xml={REPORT}".format(REPORT=args.report)]
 
-    print(pytest_cmd)
-    exit(subprocess.call(pytest_cmd))
+    # Register repos. This may take a while.
+    snactor_register(BASE_REPO)
+
+    # Find and collect leapp repositories.
+    repos = {}
+    for root, dirs, files in os.walk(BASE_REPO):
+        if ".leapp" in dirs:
+            # For more info:
+            # https://github.com/leapp-to/leapp/blob/master/leapp/snactor/commands/discover.py#L89
+            base_dir = find_repository_basedir(root)
```

Comment:
is this necessary? since you know that there is .leapp in current directory, won't find_repo_basedit always return root?